src/gl_fns.rs

```diff
@@ -19,6 +19,21 @@ impl GlFns
             ffi_gl_: ffi_gl_,
         }) as Rc<Gl>
     }
+
+    fn get_active_uniform_type(&self, program: GLuint) -> GLuint {
+        let mut size: GLint = 0;
+        let mut uniform_type: GLuint = 0;
+        unsafe {
+            self.ffi_gl_.GetActiveUniform(program,
+                                          0 as GLuint,
+                                          0 as GLsizei,
+                                          ptr::null_mut(),
+                                          &mut size,
+                                          &mut uniform_type,
+                                          ptr::null_mut());
```

Comment:
Note: according to the [spec](https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGetActiveUniform.xml), `name` is modified only if `bufSize` is more than 0 and `length` is modified only if it is not null, so it is safe to pass a null to them. But nothing is said about `size`. Looking at the [implementation](https://github.com/servo/mozangle/blob/4bfe6f83db19b1cfe7e1cf273018edcd7bee47ff/gfx/angle/checkout/src/libANGLE/Program.cpp#L1501), looks like it is always dereferenced, so we should pass something here.

Reply:
Please fix this and I'll r+ the PR.

Reply:
This doesn't need fixing. It's just a note I left about the safety of passing null pointers for easier review.

Reply:
Oh, I thought that `size` you mentioned was the NULL pointer. Never mind.